### PR TITLE
`Development`: Fix course messages e2e tests

### PR DESCRIPTION
**Pull request by asliayk**
#### General

- [X] This is a small issue that I tested locally.
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Some of the tests in `CourseMessages.spec.ts` were failing after the changes in sidebar design. The tests could not identify the newly added button for create channel/browse channels.

### Description

`browseChannelButton()`, `createChannelButton()`, and `createGroupChatButton()` have been updated so that tests can identify these buttons again.

### Steps for Testing
Check that the test runs through correctly.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **Bug Fixes**
	- Streamlined method calls for browsing channels, enhancing test clarity and maintainability.
	- Adjusted expected channel names in assertions to match new naming conventions.

- **New Features**
	- Updated UI interactions for creating channels and group chats, aligning with the new button structures. 

These enhancements improve user experience by ensuring consistent channel management and clearer test outcomes.

### CODE REVIEW

**Review by SimonEntholzer — APPROVED** [review]
All tests pass again :+1:  
https://bamboo.ase.in.tum.de/build/result/viewBuildResults.action?buildKey=ARTEMIS-AEPTMA1009-DA&buildNumber=3

File: src/test/playwright/support/pageobjects/course/CourseMessagesPage.ts [hunk 1]
```diff
@@ -18,15 +18,16 @@ export class CourseMessagesPage {
...
     }
 
     /**
      * Navigates to the channel overview section.
      */
-    browseChannelsButton(channelGroup: string) {
-        return this.page.locator(`#plusButton-${channelGroup}`);
+    async browseChannelsButton() {
+        await this.page.locator('.btn-primary.btn-sm.square-button').click();
+        await this.page.locator('button', { hasText: 'Browse Channels' }).click();
```
Comment (pzdr7) [inline src/test/playwright/support/pageobjects/course/CourseMessagesPage.ts:30]: The method name is not super intuitive to me - I'd expect the method to return a locator to the browse button (like it did in the old implementation). Could we rename it to `openBrowseChannelModal` or something similar?

File: src/test/playwright/support/pageobjects/course/CourseMessagesPage.ts [hunk 2]
```diff
@@ -18,15 +18,16 @@ export class CourseMessagesPage {
      * Clicks the button to initiate channel creation.
      */
     async createChannelButton() {
```
Comment (pzdr7) [inline src/test/playwright/support/pageobjects/course/CourseMessagesPage.ts:20]: I'd suggest the name `openChannelCreation` here

File: src/test/playwright/support/pageobjects/course/CourseMessagesPage.ts [hunk 3]
```diff
@@ -288,7 +289,8 @@ export class CourseMessagesPage {
      * Clicks the button to initiate group chat creation.
      */
     async createGroupChatButton() {
-        await this.page.locator('#plusButton-groupChats').click();
+        await this.page.locator('.btn-primary.btn-sm.square-button').click();
+        await this.page.locator('button', { hasText: 'Create Group Chat' }).click();
```
Comment (pzdr7) [inline src/test/playwright/support/pageobjects/course/CourseMessagesPage.ts:293]: Maybe `openGroupChatCreation`?

File: src/test/playwright/support/pageobjects/course/CourseMessagesPage.ts [hunk 4]
```diff
@@ -18,15 +18,16 @@ export class CourseMessagesPage {
      * Clicks the button to initiate channel creation.
      */
     async createChannelButton() {
-        await this.page.locator('#plusButton-generalChannels').click();
-        await this.page.locator('.modal-content #createChannel').click();
+        await this.page.click('.square-button > .ng-fa-icon');
```
Comment (pzdr7) [inline src/test/playwright/support/pageobjects/course/CourseMessagesPage.ts:21]: As I see it, all three buttons (create channel, browse channels, create group chat) are under the same menu and can be accessed via the same button.

Why do you use `this.page.click('.square-button > .ng-fa-icon');` here but `this.page.locator('.btn-primary.btn-sm.square-button').click()` in the other methods? Could we simply assign an ID to the button that opens the dropdown and use that to find it?

**Review by pzdr7 — CHANGES_REQUESTED** [review]
Some small comments on the code

**Review by krusche — APPROVED** [review]
Thanks for fixing the e2e tests.
I suggest to improve the code according to @pzdr7’s feedback in a follow up so that the tests pass on develop again and do not disturb any other PRs

**Pull request merged**